lxc/move.go

```diff
@@ -120,3 +146,95 @@ func (c *moveCmd) run(conf *config.Config, args []string) error {
 	del.force = true
 	return del.run(conf, args[:1])
 }
+
+// Helper to check if the container to be moved is backed by a ceph storage
+// pool, and use the special POST /containers/<name>?target=<node> API if so.
+//
+// It returns false if the container is not backed by ceph, true otherwise.
+func maybeMoveCephContaner(conf *config.Config, sourceResource, destResource, target string) (bool, error) {
+	// Parse the source.
+	sourceRemote, sourceName, err := conf.ParseRemote(sourceResource)
+	if err != nil {
+		return false, err
+	}
+
+	// Parse the destination.
+	destRemote, destName, err := conf.ParseRemote(destResource)
+	if err != nil {
+		return false, err
+	}
+
+	if sourceRemote != destRemote {
+		return false, fmt.Errorf(
+			i18n.G("You must use the same source and destination remote when using --target"))
+	}
+
+	// Make sure we have a container or snapshot name.
+	if sourceName == "" {
+		return false, fmt.Errorf(i18n.G("You must specify a source container name"))
+	}
+
+	// The destination name is optional.
+	if destName == "" {
+		destName = sourceName
+	}
+
+	// Connect to the source host
+	source, err := conf.GetContainerServer(sourceRemote)
+	if err != nil {
+		return false, err
+	}
+
+	if shared.IsSnapshot(sourceName) {
+		// TODO: implement moving snapshots.
```

Comment:
Snapshots can't be moved separately from their container, so I'm not sure what this is about.

One thing that appears to have been overlooked here though is the snapshots of the container.
If moving a container that holds a number of snapshots, all those storage volumes must also be moved to the new host.

Reply:
I see, that was overlooked indeed. I fixed the code and the tests to also handle snapshots properly, and changed the error message here accordingly.